src/Ydb.Sdk/src/Services/Table/SessionPool.cs

```diff
@@ -1,181 +1,63 @@
 ﻿using Microsoft.Extensions.Logging;
-using Ydb.Sdk.Client;
+using Ydb.Sdk.Services.Sessions;
 
 namespace Ydb.Sdk.Services.Table;
 
-public class SessionPoolConfig
-{
-    public SessionPoolConfig(
-        uint? sizeLimit = null)
-    {
-        SizeLimit = sizeLimit ?? 100;
-    }
-
-    public uint SizeLimit { get; }
-
-    internal TimeSpan KeepAliveIdleThreshold { get; } = TimeSpan.FromMinutes(5);
-    internal TimeSpan PeriodicCheckInterval { get; } = TimeSpan.FromSeconds(10);
-    internal TimeSpan KeepAliveTimeout { get; } = TimeSpan.FromSeconds(1);
-    internal TimeSpan CreateSessionTimeout { get; } = TimeSpan.FromSeconds(1);
-}
-
-internal class GetSessionResponse : ResponseWithResultBase<Session>, IDisposable
-{
-    internal GetSessionResponse(Status status, Session? session = null)
-        : base(status, session)
-    {
-    }
-
-    public void Dispose()
-    {
-        Dispose(true);
-    }
-
-    protected virtual void Dispose(bool disposing)
-    {
-        if (disposing)
-        {
-            Result.Dispose();
-        }
-    }
-}
-
-internal interface ISessionPool : IDisposable
-{
-    public Task<GetSessionResponse> GetSession();
-}
+using GetSessionResponse = GetSessionResponse<Session>;
+using NoPool = NoPool<Session>;
 
-internal class NoPool : ISessionPool
+internal sealed class SessionPool : SessionPoolBase<Session, TableClient>
 {
-    public Task<GetSessionResponse> GetSession()
+    public SessionPool(Driver driver, SessionPoolConfig config) :
+        base(
+            driver: driver,
+            config: config,
+            client: new TableClient(driver, new NoPool()),
+            logger: driver.LoggerFactory.CreateLogger<SessionPool>())
     {
-        throw new InvalidOperationException("Unexpected session pool access.");
-    }
-
-    public void Dispose()
-    {
-    }
-}
-
-internal sealed class SessionPool : ISessionPool
-{
-    private readonly Driver _driver;
-    private readonly SessionPoolConfig _config;
-
-    private readonly object _lock = new();
-    private readonly ILogger _logger;
-    private readonly TableClient _client;
-    private bool _disposed;
-
-    private readonly Dictionary<string, SessionState> _sessions = new();
-    private readonly Stack<string> _idleSessions = new();
-    private uint _pendingSessions;
-
-    public SessionPool(Driver driver, SessionPoolConfig config)
-    {
-        _driver = driver;
-        _config = config;
-
-        _logger = _driver.LoggerFactory.CreateLogger<SessionPool>();
-        _client = new TableClient(_driver, new NoPool());
-
         Task.Run(PeriodicCheck);
     }
 
-    public async Task<GetSessionResponse> GetSession()
```

Comment:
Here GetSession has changed to `Task<GetSessionResponse<TSession>>` from SessionPoolBase.
Will it break existed code?

Reply:
no, it is internal class